plugins/inputs/irqstat/irqstat.go

```diff
@@ -0,0 +1,114 @@
+package irqstat
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type Irqstat struct{}
+
+type IRQ struct {
+	ID     string
+	Type   string
+	Device string
+	Values map[string]interface{}
+}
+
+func NewIRQ(id string) *IRQ {
+	return &IRQ{ID: id, Values: make(map[string]interface{})}
+}
+
+const sampleConfig = `
+  ## To filter which IRQs to collect, make use of tagpass / tagdrop, i.e.
+  # [inputs.irqstat.tagdrop]
+    # irq = [ "NET_RX", "TASKLET" ]
+`
+
+func (s *Irqstat) Description() string {
+	return "This plugin gathers interrupts data from /proc/interrupts and /proc/softirqs."
+}
+
+func (s *Irqstat) SampleConfig() string {
+	return sampleConfig
+}
+
+func parseInterrupts(irqdata string) ([]IRQ, error) {
+	var irqs []IRQ
+	var cpucount int
+	scanner := bufio.NewScanner(strings.NewReader(irqdata))
+	for scanner.Scan() {
+		var irqval, irqtotal int64
+		var irqtype, irqdevice string
+		fields := strings.Fields(scanner.Text())
+		if len(fields) > 0 && fields[0] == "CPU0" {
```

Comment:
I think you should scan the first line outside of the loop.  This will reduce the complexity of this loop and prevent any strange effects if this matches somehow.